### ISSUE TITLE
[Bug] Overactive warning: The parent file cannot be found in the project file

### ISSUE THREAD

**Issue by mfeingol** [xamarin/Xamarin.Forms#9167]
### Description

I'm using a shared project to build a Xamarin.Forms app. I have iOS and Android projects that use the shared project. Both projects display the following warning for several xaml/cs file pairs in the VS Error List.

```
Warning		The parent file, 'SettingsPage.xaml', for file 'C:\Operations\Build\Home\Sideroads\Source\Sideroads.Xamarin\Sideroads.Shared\Views\SettingsPage.xaml.cs' cannot be found in the project file.
```

Needless to say, the files in question exist.

Oddly, the warning is not displayed for all pairs, just some. I am unable to see a difference between a pair that warns and a pair that doesn't. The following are extracts from the .projitems file:

Warning:

```
    <Compile Include="$(MSBuildThisFileDirectory)Views\SettingsPage.xaml.cs">
      <DependentUpon>SettingsPage.xaml</DependentUpon>
      <SubType>Code</SubType>
    </Compile>
...
  <ItemGroup>
    <EmbeddedResource Include="$(MSBuildThisFileDirectory)Views\SettingsPage.xaml">
      <SubType>Designer</SubType>
      <Generator>MSBuild:UpdateDesignTimeXaml</Generator>
    </EmbeddedResource>
  </ItemGroup>
```

No warning:

```
    <Compile Include="$(MSBuildThisFileDirectory)Views\ShellPage.xaml.cs">
      <DependentUpon>ShellPage.xaml</DependentUpon>
      <SubType>Code</SubType>
    </Compile>

...

  <ItemGroup>
    <EmbeddedResource Include="$(MSBuildThisFileDirectory)Views\ShellPage.xaml">
      <SubType>Designer</SubType>
      <Generator>MSBuild:UpdateDesignTimeXaml</Generator>
    </EmbeddedResource>
  </ItemGroup>
```

### Steps to Reproduce

Unclear. I see it 100% in my project. I don't see it in a vanilla XF project generated by VS.

### Expected Behavior

No warnings.

### Actual Behavior

Warnings for a number of xaml/.cs pairs, but not all. No visible difference in csproj/projitems between the ones that warn and the ones that don't.

### Basic Information

- Version with issue: 4.4
- Last known good version: Unknown, but I started seeing this relatively recently
- IDE: VS 2019 16.4.2

**Comment by hartez** [comment]
Any chance you could upload a small project that displays this behavior?

**Comment by mfeingol** [comment]
@hartez: I've tried removing meat from my project, but somewhere along the way the warnings just disappear.

I can add you to my project on GitHub, if that would help.

**Comment by hartez** [comment]
@mfeingol Sure, add me and I'll take a look.

**Comment by mfeingol** [comment]
You're in: https://github.com/mfeingol/sideroads

**Comment by hartez** [comment]
Okay, I'm definitely seeing those warnings. That's … weird.

Randomly, I tried moving one of the item groups to another part of the projitems file and the warnings went away. If you grab 

```
  <ItemGroup>
    <EmbeddedResource Include="$(MSBuildThisFileDirectory)Views\SettingsPage.xaml">
      <SubType>Designer</SubType>
      <Generator>MSBuild:UpdateDesignTimeXaml</Generator>
    </EmbeddedResource>
  </ItemGroup>
```

and cut/paste it as the last item group in the file, then reload the solution - do you still see the errors? 

(Not proposing this as a "solution", just want to see if the behavior is consistent.)

**Comment by mfeingol** [comment]
@hartez: thanks for taking a look. It is pretty weird.

What I'm seeing is that the warnings go away as you describe, but return when VS is closed and the solution is reopened. Are you seeing the same thing?

**Comment by daves1992** [comment]
![parent file](https://user-images.githubusercontent.com/29727003/73181453-289e4f80-410f-11ea-8491-725b137a3fa4.png)

I have had this issue for a little while too in a couple of my projects. Weird how it's not for all the files.

**Comment by mfeingol** [comment]
@hartez: any luck figuring this out?

**Comment by samhouts** [comment]
@mfeingol I don't think this is due to the Xamarin.Forms SDK, so we should get it submitted to the VS team. Can you please report this via send feedback within Visual Studio? Thanks!

**Comment by Erarndt** [comment]
@mfeingol I'm beginning to look into this, but I'm not sure how to repro this. What can I do to repro the issue?

**Comment by artemious7** [comment]
Has anyone submitted this to the VS Team? Please, paste a link to the ticket.
Still having this issue.

**Comment by mfeingol** [comment]
This is not fixed in VS 16.8.2.

**Comment by Draskown** [comment]
Although I'm running a project in WinForms, the issue is still relevant :\.
